src/mlpack/tests/feedforward_network_test.cpp

```diff
@@ -410,4 +410,134 @@ BOOST_AUTO_TEST_CASE(DropConnectNetworkTest)
       (dataset, labels, dataset, labels, 2, 10, 50, 0.2);
 }
 
-BOOST_AUTO_TEST_SUITE_END();
\ No newline at end of file
+/**
+ * Train and evaluate a Maxout network(with a baselayer) with the
+ * specified structure.
+ */
+template<typename MatType = arma::mat>
+void BuildMaxoutNetwork(MatType& trainData,
+                        MatType& trainLabels,
+                        MatType& testData,
+                        MatType& testLabels,
+                        const size_t outputSize,
+                        const size_t maxoutHiddenLayerSize,
+                        const size_t maxoutUnitSize,
+                        const size_t maxEpochs,
+                        const double classificationErrorThreshold)
+{
+ /*
+  *  Construct a feed forward network with trainData.n_rows input nodes,
+  *  maxoutUnitSize hidden nodes and trainLabels.n_rows output nodes. The
+  *  network struct that looks like:
+  *
+  *  Input         Maxout         Output
+  *  Layer         Layer          Layer
+  * +-----+       +-----+        +-----+
+  * |     |       |     |        |     |
+  * |     +------>|     +------->|     +
+  * |     |     +>|     |      +>|     |
+  * +-----+     | +--+--+      | +-----+
+  *             |              |
+  *   Bias      |   Bias       |
+  *   Layer     |   Layer      |
+  *  +-----+    |  +-----+     |
+  *  |     |    |  |     |     |
+  *  |     +----+  |     +-----+
+  *  |     |       |     |
+  *  +-----+       +-----+
+  *
+  *
+  */
+
+  FFN<NegativeLogLikelihood<> > model;
+  model.Add<Maxout<> >(trainData.n_rows, maxoutHiddenLayerSize, maxoutUnitSize);
+  model.Add<Linear<> >(maxoutUnitSize, outputSize);
+  model.Add<LogSoftMax<> >();
+
+  RMSprop<decltype(model)> opt(model, 0.01, 0.88, 1e-8,
+      maxEpochs * trainData.n_cols, -1);
+
+  model.Train(std::move(trainData), std::move(trainLabels), opt);
+
+  MatType predictionTemp;
+  model.Predict(testData, predictionTemp);
+  MatType prediction = arma::zeros<MatType>(1, predictionTemp.n_cols);
+
+  for (size_t i = 0; i < predictionTemp.n_cols; ++i)
+  {
+    prediction(i) = arma::as_scalar(arma::find(
+        arma::max(predictionTemp.col(i)) == predictionTemp.col(i), 1)) + 1;
+  }
+
+  size_t error = 0;
+  for (size_t i = 0; i < testData.n_cols; i++)
+  {
+    if (int(arma::as_scalar(prediction.col(i))) ==
+        int(arma::as_scalar(testLabels.col(i))))
+    {
+      error++;
+    }
+  }
+
+  double classificationError = 1 - double(error) / testData.n_cols;
+  BOOST_REQUIRE_LE(classificationError, classificationErrorThreshold);
+}
+
+/**
+ * Train the maxout network on a larger dataset.
+ */
+BOOST_AUTO_TEST_CASE(MaxoutNetworkTest)
```

Comment:
Hi @zoq I decresed number of iterations by fraction of 10 (7 and 5 respectively) and test is successful with less iterations. Here is output of `time`.
```
real	0m10.143s
user	0m11.111s
sys	0m24.168s
```